pkg/apis/pingcap/v1alpha1/dmcluster.go

```diff
@@ -245,6 +245,48 @@ func (dc *DMCluster) MasterIsAvailable() bool {
 		return false
 	}
 
+	if dc.Status.Master.Phase == SuspendPhase {
+		return false
+	}
+
+	return true
+}
+
+func (dc *DMCluster) ComponentIsNormal(typ MemberType) bool {
+	status := dc.ComponentStatus(typ)
+	if status == nil {
+		return false
+	}
+	return status.GetPhase() == NormalPhase
+}
+
+func (dc *DMCluster) ComponentIsSuspending(typ MemberType) bool {
+	status := dc.ComponentStatus(typ)
+	if status == nil {
+		return false
+	}
+	return status.GetPhase() == SuspendPhase
+}
+
+func (dc *DMCluster) ComponentIsSuspended(typ MemberType) bool {
+	spec := dc.ComponentSpec(typ)
+	status := dc.ComponentStatus(typ)
+	if spec == nil || status == nil {
+		return false
+	}
+
+	if !dc.ComponentIsSuspending(typ) {
+		return false
+	}
+
+	action := spec.SuspendAction()
+	if action != nil && action.SuspendStatefulSet {
+		if status.GetStatefulSet() != nil {
+			// the statefulset is set to nil by suspender when the sts is deleted.
+			return false
+		}
+	}
+
 	return true
```

Comment:
The default value should be false, right?

Reply:
yes, return true when acton is set but no resources need to suspend.

Reply:
But here, it will return true for the following cases:
1. action==nil
2. action.SuspendStatefulSet==false
3. action.SuspendStatefulSet==true && status.StatefulSet==nil
Only case 3 should return true, because the cluster is not suspended for cases 1&2, right?